src/compiler/parser.ts

```diff
@@ -83,7 +83,8 @@ module ts {
                     visitNodes(cbNodes, node.modifiers) ||
                     visitNodes(cbNodes, (<SignatureDeclaration>node).typeParameters) ||
                     visitNodes(cbNodes, (<SignatureDeclaration>node).parameters) ||
-                    visitNode(cbNode, (<SignatureDeclaration>node).type);
+                    visitNode(cbNode, (<SignatureDeclaration>node).type) ||
+                    visitNode(cbNode, (<SignatureDeclaration>node).typePredicate);

```

Comment:
I was thinking it would be stored separately on the Signature, but not separately on the SignatureDeclaration. It is semantically separate from the return type, but syntactically not.